### ISSUE TITLE
Refractor losses class

### ISSUE THREAD

**Issue by ViktorKaz** [alan-turing-institute/mlaut#22]
[mlaut.losses](https://github.com/kiraly-group/mleap/blob/master/mlaut/analyze_results/losses.py) takes `metric` as string attribute. This should be refractored to take class instance instead (sklearn.metrics  accuracy_score, mean_squared_error, etc).
